onnx/shape_inference/implementation.cc

```diff
@@ -241,11 +240,9 @@ static void InferShapesImpl(
       }
       for (int i = 0; i < n.output_size(); ++i) {
         const auto* inferredType = ctx.getOutputType(i);
-        if (!inferredType->has_tensor_type() &&
-            !inferredType->has_sequence_type()) {
+        if (inferredType->value_case() == TypeProto::ValueCase::VALUE_NOT_SET) {
```

Comment:
Great, thanks! Can we also eliminate the lines 246 to 254? It will help simplify the code and make it easier to maintain and extend in the future.